spec.html

```diff
@@ -7620,7 +7630,7 @@ <h1>SetFunctionLength ( _F_, _length_ )</h1>
       <emu-alg>
         1. Assert: _F_ is an extensible object that does not have a `"length"` own property.
         1. Assert: Type(_length_) is Number.
-        1. Assert: _length_ &ge; 0 and ! IsInteger(_length_) is *true*.
+        1. Assert: ! IsNonNegativeInteger(_length_) is *true*.
```

Comment:
There's no reason to prefix all of these `IsNonNegativeInteger` uses with `!`. Let's drop them all, please.

Reply:
Thank you for review, please see https://github.com/tc39/ecma262/pull/1529#discussion_r281878951.

If it was up to me, I wouldn't prefix abstract ops that never return abrupt completions with `!`.

Reply:
There's no completion records involved here, though. That is, unless you're of the camp that believes all abstract operations return completion records. I guess we'll clear that up with #497, so what we do here doesn't really matter to me. I just think it's ridiculous to litter the spec with `!` when the abstract op being called neither explicitly creates a completion record nor returns a completion record created by another abstract op.

Reply:
It's not a camp; they currently *do* all return completion records - so pending #497 or another change, the ! are needed here.

Reply:
I'll just leave this here: https://github.com/tc39/ecma262/issues/496#issuecomment-491603203